enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,751 @@
+Cluster Autoscaler has made a change to a Cluster API resource and then sync
+the change to the corresponding Machine API resource, regardless of which resource
+is authoritative. The scope of API changes from the Cluster Autoscaler are limited
+to updating the scale subresource and annotations of MachineSet resources.
+
+Update the Cluster Autoscaler Operator to be namespace aware. This requires
+changing the operator to recognize when it has a Machine API or Cluster API
+reference in a MachineAutoscaler, and then use the appropriate namespace to
+locate the resource. The Cluster API MachineSet sync controller will be updated
+to ensure that when the Cluster Autoscaler Operator adds the autoscaling
+annotations that they are copied to any related resources, regardless of which
+is authoritative.
+
+Update the Cluster API MachineSet sync controller to recognize the
+scale-from-zero annotations and copy them from the Machine API resources to the
+Cluster API resources. This ensures consistent data representation on both
+the authoritative and non-authoritative records.
+
+Leave the Cluster Autoscaler and Cluster Autoscaler Operator in the
+openshift-machine-api namespace. This will continue the user experience for
+managing autoscaling.
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for managing MachineSet,
+ClusterAutoscaler, and MachineAutoscaler resources in a cluster.
+
+1. The cluster administrator creates a Cluster API MachineSet named "scaling-set-1"
+  intended for autoscaling.
+2. The cluster administrator creates a ClusterAutoscaler resource to deploy
+  the cluster autoscaler component in the cluster.
+3. The cluster adiminstrator creates a MachineAutoscaler resource referencing the
+  "scaling-set-1" MachineSet in the `scaleTargetRef` field.
+4. The cluster administrator can see from logs and metadata on the MachineSet
+  that it is being evaluated by the cluster autoscaler for autoscaling.
+
+In general, the workflow for users should continue to meet the current expectations
+for cluster autoscaler functionality. The main change to the previous workflows
+is the inclusion of Cluster API MachineSets (e.g. MachineSets that have the
+resource group `cluster.x-k8s.io`) in MachineAutoscaler resources.
+
+### API Extensions
+
+This enhancement will require a change to the `status` field of the MachineAutoscaler
+API type. There is no expected change to the `spec` field, or the behavior that
+users expect.
+
+The Cluster Autoscaler Operator will be changed to include logic that can detect
+the API group for any MachineSet that is referenced in the `scaleTargetRef` field
+of MachineAutoscaler resources. The change will instruct the Operator to search
+for records in the `openshift-cluster-api` namespace for resources with the
+`cluster.x-k8s.io` group, and to search in the `openshift-machine-api` namespace
+for resource with the `machine.openshift.io` group.
```

Comment:
Every resource in the `openshift-machine-api` namespace will have a mirror in the `openshift-cluster-api` namespace, as such, I'd expect searching the `openshift-cluster-api` namespace to be enough. There's no need to look at the MAPI objects is there?